unitary/examples/quantum_rpg/qaracter.py

```diff
@@ -0,0 +1,140 @@
+  their qubits measured and at least half are measured as one.
+  If neither of these is true, the Qaracter is considered to
+  be ACTIVE.
+
+  Additional HP can be added with the add_hp() function.
+  Additional quantum effects can be added with the add_quantum_effects()
+  function.
+
+  Hit point values can be sampled (a random result pulled from
+  the simulated distribution without measuring, which is possible
+  only in simulation) with sample(hp_num, False).
+
+  Hit point values can be measured by calling sample(hp_num, True).
+
+
+  Args:
+      name: name of this character as a string.
+  """
+
+    def __init__(self, name: str):
+        super().__init__()
+        self.name = name
+        self.health_status: Dict[alpha.QuantumObject, int] = {}
+        self.level = 0
+        self.add_hp()
+
+    def is_npc(self) -> bool:
+        """Returns Trus if a non-player or False if a player.
+
+    Inheritors of NPCs should override this function.
+    """
+        return False
+
+    def quantum_object_name(self, hp_num: int) -> str:
+        """Canonical name of a QuantumObject for this Qaracter.
+
+      Represents the n-th HP of a Qaracter.
+
+      Args:
+          hp_num: Number of the health point requested.  Health points
+              start at 1 and go up to the Qaracter's level (inclusive).
+      """
+        return f'{self.name}_{hp_num}'
+
+    def add_hp(self) -> alpha.QuantumObject:
+        """Adds an additional Health Point (HP to the Qaracter.
+
+      Each HP added to the Qaracter is an additional QuantumObject
+      with no effects (i.e. is a qubit with no operations).
+      Adding a HP also increases the Qaracter's level.
+      """
+        self.level += 1
+        obj = alpha.QuantumObject(self.quantum_object_name(self.level),
+                                  enums.HealthPoint.HURT)
+        self.add_object(obj)
+        return obj
+
+    def get_hp(self, hp_name: str) -> alpha.QuantumObject:
+        """Retrieves a QuantumObject with the specified name."""
+        for obj in self.objects:
+            if hp_name == obj.name:
+                return obj
+
+    def active_qubits(self) -> List[str]:
+        """Returns the names of all active (non-measured) quantum objects (HP)."""
+        return [
+            self.quantum_object_name(i)
+            for i in range(1, self.level + 1)
+            if self.quantum_object_name(i) not in self.health_status
+        ]
+
+    def add_quantum_effect(self, effect, quantum_name: Union[int, str]):
+        """Adds a Quantum Effect to a specific quantum object."""
+        if isinstance(quantum_name, int):
+            quantum_name = self.quantum_object_name(quantum_name)
+        hp_obj = self.get_hp(quantum_name)
+        effect(hp_obj)
+
+    def is_down(self) -> bool:
+        """Returns True if over half of the HP have been measured as `HURT`."""
+        damage = len(
+            [q for q in self.health_status if self.health_status[q] == 0])
```

Comment:
`sum(status == 0 for status in self.health_status.values())` would be cleaner (here and for `is_escaped`)

Reply:
Good call, done.